spec/features/service_providers_spec.rb

```diff
@@ -21,6 +21,7 @@
       select user.teams[0].name, from: 'service_provider_group_id'
       select 'IAL2', from: 'Identity Assurance Level (IAL)'
       select 'AAL2', from: 'Authentication Assurance Level (AAL)'
+      attach_file 'cert', Rails.root.join('spec', 'fixtures', 'cert.crt')
```

Comment:
I added a `build_cert` helper to help generate valid PEMs, so if we wanted to write a tempfile rather than check these in, we could do something like:

```suggestion
      Tempfile.create(['cert', '.crt']) do |tmp|
        File.write(tmp.path, build_cert)
        attach_file 'cert', tmp.path
      end
```

Reply:
Ooh good call love it - thanks!